tests/test_algorithm.py

```diff
@@ -425,6 +425,23 @@ def test_asset_lookup(self):
         self.assertIsInstance(algo.sid(3), Equity)
         self.assertIsInstance(algo.sid(4), Equity)
 
+        # Supplying a non-string argument to future_symbol()
+        # should result in a TypeError.
+        with self.assertRaises(TypeError):
+            algo.symbol(1)
+
+        with self.assertRaises(TypeError):
+            algo.symbol((1,))
+
+        with self.assertRaises(TypeError):
+            algo.symbol(set(1,))
+
+        with self.assertRaises(TypeError):
+            algo.symbol(list(1,))
```

Comment:
To be clear, these expressions aren't even making it into the `symbol` call.  They're failing in the calls to `list` and `set`.